nucleus/deploy/client.py

```diff
@@ -161,22 +170,31 @@ def create_model_bundle(
 
         bundle_metadata = {}
         # Create bundle
-        if model is not None:
-            bundle = dict(model=model, load_predict_fn=load_predict_fn)
+        if predict_fn_or_cls:
+            bundle_fn_or_cls = predict_fn_or_cls
+            serialized_bundle = cloudpickle.dumps(bundle_fn_or_cls)
+            if inspect.isfunction(predict_fn_or_cls):
+                source_code = inspect.getsource(predict_fn_or_cls)
+            else:
+                source_code = inspect.getsource(predict_fn_or_cls.__class__)
+            bundle_metadata["predict_fn_or_cls"] = source_code
+        elif model is not None:
+            bundle_func_1 = dict(model=model, load_predict_fn=load_predict_fn)
+            serialized_bundle = cloudpickle.dumps(bundle_func_1)
             bundle_metadata["load_predict_fn"] = inspect.getsource(
-                load_predict_fn
+                load_predict_fn  # type: ignore
             )
         else:
-            bundle = dict(
+            bundle_func_2 = dict(
```

Comment:
Can we just call both of these `bundle_func`? These branches are mutually exclusive, so even Python's funky dynamic scoping rules wouldn't be an issue.

Reply:
yeah, I did it for typing check that a linter performs. It complains that a variable `bundle` has different types in different branches, i.e dict/func/class_obj
I will play with the Optional[Dict...]